src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -181,6 +181,31 @@ function validatePropTypes(element) {
   }
 }
 
+/**
+ * Given an element, freeze `type` and `defaultProps` (Only used in DEV)
+ *
+ * @param {ReactElement} element
+ */
+function freezeTypeAndDefaultProps(element) {
+  var type = element.type;
+  if (typeof type !== 'function' || type.wasTaggedReactComponent) {
+    return;
+  }
+  if (canDefineProperty) {
+    Object.defineProperty(type, 'wasTaggedReactComponent', { value: true });
+    Object.defineProperty(type, 'defaultProps', {
+      writable: false,
+      configurable: false,
+    });
+  } else {
+    type.wasTaggedReactComponent = true;
+  }
+  if (Object.freeze) {
+    Object.freeze(type);
```

Comment:
@sebmarkbage revised in 58cd9bc560be8d088dc12ba8995c2170a83b08a0:
 - dropped freeze for `element.type` (only default props)
 - updated function name to `freezeDefaultProps`
 - updated tests to reflect changes

thank you for your review and guidance :)